### PR TITLE
Add global message functionality

### PR DESCRIPTION
**Pull request by h-lame**
For: https://trello.com/c/W8bleowe/296-add-global-message-to-release-app

We introduce a global singleton Site object for storing global settings.  The only global setting so far is a `status_notes` field (to mirror the existing one on `Application`).  If this is present it will be shown on the application show and deploy pages.

The goal here is to provide a mechanism to warn developers about deploy freezes.  We have planned deploy freezes that everyone should be aware of because we do some messaging beforehand (e.g. Christmas holidays, or the budget), but we also have ad-hoc ones we may put in place during incidents and developers may miss these announcements.

See the commits for the blow-by-blow details.

### CODE REVIEW

**Comment by dwhenry** [comment]
@h-lame would it make sense to have a secondary flag around `block non-urgent deploys`, that would be required to have an `I want to deploy anyway` checkbox that was validated server-side. That would avoid issues with deploying from a stale version of the app. 

This might be out of scope for this change. :)

**Comment by h-lame** [comment]
@dwhenry - we spoke about this when discussing doing this and decided it was a useful feature, but not one to implement at the same time as this message feature.

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -15,6 +15,7 @@
   <%= nav_link 'Applications', applications_path %>
   <%= nav_link 'Activity', activity_path %>
   <%= nav_link 'Archived', archived_applications_path %>
+  <% if current_user.may_deploy? %><%= nav_link 'Site settings', site_path %><% end %>
```
Comment (dwhenry) [inline app/views/layouts/application.html.erb]: Don't know if this needs three blocks, but that is more of a style comment than a required change

```
<%= nav_link('Site settings', site_path) if current_user.may_deploy? %>
```

Comment (h-lame) [reply app/views/layouts/application.html.erb]: Of course, silly me.  I'll fix this in a rebase and then merge.  Thanks for the review!

**Pull request merged**